### PR TITLE
[jvm-packages] enable predictLeaf/predictContrib/treeLimit in 0.8 

### PR DESCRIPTION
**Pull request by CodingCat**
missing the test cases

### CODE REVIEW

**Comment by hcho3** [comment]
@CodingCat Should we shoot for merging this PR before releasing 0.80? If we need to, we can delay 0.80 release a bit.

**Comment by CodingCat** [comment]
@yanboliang would you please take a look?

File: jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala [hunk 1]
```diff
@@ -313,7 +315,79 @@ class XGBoostClassificationModel private[ml](
...
+  }
+
+  private def produceResultIterator(
+      originalRowItr: Iterator[Row],
+      rawPredictionItr: Iterator[Row],
+      probabilityItr: Iterator[Row],
+      predLeafItr: Iterator[Row],
+      predContribItr: Iterator[Row]): Iterator[Row] = {
+    // the following implementation is to be improved
+    if (isDefined(leafPredictionCol) && isDefined(contribPredictionCol)) {
```
Comment (yanboliang) [inline jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala]: I have a little concern that users may set ```leafPredictionCol``` with ```""```, their intention should be not to output this column, but here we don't handle this case. At MLlib, we check ```$(leafPredictionCol).isEmpty``` as well. What do you think about it?

Comment (CodingCat) [reply jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostClassifier.scala]: yes, good point, let me revise it

**Comment by CodingCat** [comment]
@yanboliang please check the latest update

**Comment by yanboliang** [comment]
Looks good, it seams the failure is not relevant.

**Comment by hcho3** [comment]
I fixed the error in the Jenkins build. It should build successfully now.

**Comment by CodingCat** [comment]
@hcho3 I merged this and cherry-picked to release_0.80 branch

**Comment by hcho3** [comment]
@CodingCat Okay, I will do a code freeze using this commit. The branch will be re-based against the master at this commit.

As for the doc building, I'm going to place the build script on the CI server. The built doc will be placed to an S3 bucket, which will be then downloaded by ReadTheDocs. Expect a commit by end of today.

**Pull request merged**